src/azure-cli/azure/cli/command_modules/role/custom.py

```diff
@@ -1659,7 +1658,7 @@ def _build_key_credentials(key_value=None, key_type=None, key_usage=None,
         start_date = dateutil.parser.parse(start_date)
 
     if not end_date:
-        end_date = start_date + relativedelta(years=1) - relativedelta(hours=24)
+        end_date = start_date + datetime.timedelta(days=_years_to_days(1))
```

Comment:
The removal of `- relativedelta(hours=24)` is to address https://github.com/Azure/azure-cli/pull/13769#issuecomment-1980486876